touchview/src/main/java/com/ortiz/touchview/TouchImageView.java

```diff
@@ -537,7 +537,7 @@ public void setZoom(TouchImageView img) {
     public PointF getScrollPosition() {
         Drawable drawable = getDrawable();
         if (drawable == null) {
-            return null;
+            return new PointF(.5F, .5F);
```

Comment:
It's not wrong to center it, but is't it more restrict when we use `@NonNull` especailly when you read the comment ? 

```
/**
     * Set zoom parameters equal to another TouchImageView. Including scale, position,
     * and ScaleType.
     */
public void setZoom(@NonNull TouchImageView img) {
```
I've no strong opinion

Reply:
Not sure if I understand you correct, so let me rephrase.
The view is not the problem, it's the **drawable** of that view being null, causing the crash.

If you are suggesting to change `setZoom(TouchImageView img)` to `setZoom(@NonNull TouchImageView img)` -> that will **not** guarantee `getDrawable()` returning a NonNull value. So, that annotation will not prevent a crash.

I agree that enforcing a valid view seems the best option, but I don't have suggestions for enforcing a valid view. E.g. I have no ideas on how to enforce a view with a NonNull drawable.